repos/system_upgrade/el7toel8/actors/targetuserspacecreator/libraries/userspacegen.py

```diff
@@ -47,29 +50,45 @@ def _prep_repository_access(context, target_userspace):
     context.copytree_from('/etc/yum.repos.d', os.path.join(target_userspace, 'etc', 'yum.repos.d'))
 
 
-def get_os_variant():
-    """
-    Retrieves the OS variant from the first matching OSReleaseFacts message. Returns an empty string otherwise.
-    """
-    for msg in api.consume(OSReleaseFacts):
-        if msg.variant_id:
-            return msg.variant_id
-    return ''
-
-
 def _get_product_certificate_path():
     """
     Retrieves the required / used product certificate for RHSM.
     """
-    sys_var = get_os_variant()
-    var_prodcert = {'server': '479.pem'}
-    if sys_var not in var_prodcert:
-        raise StopActorExecutionError(
-            message=('Failed to to retrieve Product Cert file.'
-                     'Product cert file not available for System Variant \'{}\'.'.format(sys_var))
-        )
+    config = next(api.consume(IPUConfig), None)
```

Comment:
This is wrong. You do not need to consume IPUConfig - that's not how the configuration works.

Reply:
to expand on this. You could do: 

`-
api.current_actor().configuration.<model_field_you_want>`